Update docs for upgrade to React Query v3 (#424)
blitz-js/blitzjs.com: app/pages/docs/testing.mdx

- [@testing-library/jest-dom](https://testing-library.com/docs/ecosystem-jest-dom)
- [@testing-library/react](https://testing-library.com/docs/react-testing-library/intro)
- [@testing-library/react-hooks](https://react-hooks-testing-library.com/)

## Testing Utilities {#testing-utilities}

Blitz provides some basic testing utilities in the generated
`test/utils.tsx` file. These are intended for use with Jest and React
Testing Library to allow advanced configuration for your testing
environment. Customizing this file can be useful for testing things like
global providers as well as certain parts of your app which Blitz handles
internally such as routing and data querying.

### `BlitzProvider` {#blitz-provider}

This component works as the global provider for data queries and mutations
in your tests. Internally, it wraps the `QueryClient` component from
`react-query` and adds some additional configuration options.

`BlitzProvider` accepts the following props:

> - `client` - The global query client used by React Query. The default
>   `queryClient` exported from `blitz` can be used here.

Reviewer: ```suggestion
- `client` - The global query client used by React Query. By default the
  `queryClient` exported from `blitz` is used unless you override it.
```
